### PR TITLE
docs: fix custom role definition & SP creation

### PR DESCRIPTION
**Pull request by carlosbustillordguez**
This PR fixes the Azure Custom Role definition and creates properly the Service Principal.

For the Service Principal creation, when `--scopes` is set we need to specify both `--role` and `--scopes`. This can be reproduced by issuing:

```bash
az ad sp create-for-rbac --name "velero" --query 'password' -o tsv \
  --scopes  /subscriptions/$AZURE_SUBSCRIPTION_ID
```

and we should get:

```bash
Usage error: To create role assignments, specify both --role and --scopes.
```

### CODE REVIEW

**Comment by blackpiglet** [comment]
@carlosbustillordguez 
Could you follow the guideline to resolve the DCO check error?
https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/172/checks?check_run_id=11964719947

**Comment by carlosbustillordguez** [comment]
> @carlosbustillordguez Could you follow the guideline to resolve the DCO check error? https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/172/checks?check_run_id=11964719947
Hi @blackpiglet, done!

**Comment by blackpiglet** [comment]
Is #173 also related to this PR?
Looks like their share some same modifications.

**Comment by carlosbustillordguez** [comment]
> Is #173 also related to this PR? Looks like their share some same modifications.

It seems #173 detected some of the errors in the doc that I detected.

**Comment by ywk253100** [comment]
@carlosbustillordguez Yes, please rebase your commit and resolve the conflict.
Is a new version of Azure CLI needed for the new changes just like mentioned [here](https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/171#issuecomment-1460374284)?

**Comment by carlosbustillordguez** [comment]
> @carlosbustillordguez Yes, please rebase your commit and resolve the conflict. Is a new version of Azure CLI needed for the new changes just like mentioned [here](https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/171#issuecomment-1460374284)?

Hi @ywk253100! I just rebased my commit and resolved the conflicts. Regarding the Azure CLI version, I am using the following:
```bash
{
  "azure-cli": "2.45.0",
  "azure-cli-core": "2.45.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "azure-devops": "0.26.0"
  }
}
I think we should into account that version (2.45.0).
```

File: README.md [hunk 1]
```diff
@@ -191,27 +192,37 @@ There are two ways to specify the role: use the built-in role or create a custom
...
           "Microsoft.Compute/disks/beginGetAccess/action",
           "Microsoft.Compute/snapshots/read",
           "Microsoft.Compute/snapshots/write",
-          "Microsoft.Compute/snapshots/delete",
-          "Microsoft.Storage/storageAccounts/listkeys/action",
-          "Microsoft.Storage/storageAccounts/regeneratekey/action"
+          "Microsoft.Compute/snapshots/delete"
       ],
+      "notActions": [],
+      "dataActions": [
```
Comment (ywk253100) [inline README.md:215]: After reverting some changes in https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/173, we don't need the actions listed in the `dataAction` part, so seems we don't need to make this change anymore.

**Comment by ywk253100** [comment]
@carlosbustillordguez Your changes are based on the code introduced by https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/111 which contains several mistakes.

In #173, we revert the changes made by #111, so seems most of the changes introduced by this PR aren't needed anymore.

File: README.md (hunk 1 above)
Comment (carlosbustillordguez) [reply README.md:215]: ok, I see... I just followed the instructions on `master` branch and works in Azure.

**Pull request closed without merging**